Demo/Demo/FeatureViewControllers/CardDemoViewController.swift

```diff
@@ -86,8 +89,88 @@ class CardDemoViewController: FeatureBaseViewController {
+        let new = current.replacingCharacters(in: changedRange, with: string)
+
+        if textField == cardNumberTextField {
+            formatCardNumber(textField: textField, newString: new)
+            toggleButton(cardNumber: textField.text ?? "", expirationDate: expirationDate, cvv: cvv)
+            setCursorLocation(textField: textField, string: string, range: range)
+            return false
+        } else if textField == expirationTextField {
+            formatExpirationDate(textField: textField, newString: new)
+            toggleButton(cardNumber: cardNumber, expirationDate: textField.text ?? "", cvv: cvv)
+            setCursorLocation(textField: textField, string: string, range: range)
+            return false
+        } else if textField == cvvTextField {
+            /// Limit cvv character count to be 4 characters max
+            guard new.count <= 4 else {
+                return false
+            }
+
+            textField.text = new
+            toggleButton(cardNumber: cardNumber, expirationDate: expirationDate, cvv: textField.text ?? "")
+
+            return false
+        }
+        return true
+    }
+
     @objc func didTapPayButton() {
         updateTitle("Processing order...")
-        // TODO: Call `processOrder`
+        payButton.startAnimating()
+        // TODO: Call `processOrder` and `payButton.stopAnimating()` once process order is complete
+    }
+
+    private func formatCardNumber(textField: UITextField, newString: String) {
+        /// clean the text field test to remove spaces
+        let cleanedText = newString.replacingOccurrences(of: " ", with: "")
+
+        /// check that the count of the text field is less than or equal to the card types max length
+        guard cleanedText.count <= CardType.unknown.getCardType(cleanedText).maxLength else { return }
+
+        textField.text = cardFormatter.formatCardNumber(cleanedText)
+    }
+
+    private func formatExpirationDate(textField: UITextField, newString: String) {
+        /// clean the text field text to remove slash and spaces
+        let cleanedText = newString.replacingOccurrences(of: "/", with: "").replacingOccurrences(of: " ", with: "")
+
+        /// limit expiration date to 4 characters max
+        guard cleanedText.count <= 4 else { return }
+
+        textField.text = cardFormatter.formatExpirationDate(cleanedText)
+    }
+
+    private func setCursorLocation(textField: UITextField, string: String, range: NSRange) {
+        /// Holders for the cursor positions
+        let positionOriginal = textField.beginningOfDocument
+        let cursorLocation = textField.position(from: positionOriginal, offset: (range.location + NSString(string: string).length))
+
+        if let cursorLocation = cursorLocation {
+            textField.selectedTextRange = textField.textRange(from: cursorLocation, to: cursorLocation)
+        }
+    }
+
+    private func toggleButton(cardNumber: String, expirationDate: String, cvv: String) {
```

Comment:
Naming nitpick - `toggle button` is a bit confusing. Maybe `enableButton` or `enableButtonForClick`?